api/src/main/java/io/opencensus/stats/MeasureMap.java

```diff
@@ -51,6 +52,17 @@
    */
   public abstract MeasureMap put(MeasureLong measure, long value);
 
+  /**
+   * Associate the contextual information of an {@code Exemplar} to this {@link MeasureMap}.
+   *
+   * <p>If this method is called multiple times, only the last string map will be kept.
+   *
+   * @param attachments contextual information of an {@code Exemplar}.
+   * @return this
+   * @since 0.16
+   */
+  public abstract MeasureMap withAttachments(Map<String, String> attachments);
```

Comment:
If we use "attachments" in the method name, I think we should use that wording in opencensus-specs, to ensure consistency across APIs.  I don't mind merging this PR first, though.

Reply:
Yes, I'll update the method names in specs in a following PR.